packages/mixin-get-child-by-name/src/index.ts

```diff
@@ -11,20 +11,43 @@ DisplayObject.prototype.name = null;
 /**
  * Returns the display object in the container.
  *
+ * Recursive searches are done in a preorder traversal.
+ *
  * @method getChildByName
  * @memberof PIXI.Container#
  * @param {string} name - Instance name.
+ * @param {boolean}[deep=false] - Whether to search recursively
  * @return {PIXI.DisplayObject} The child with the specified name.
  */
-Container.prototype.getChildByName = function getChildByName(name: string): DisplayObject
+Container.prototype.getChildByName = function getChildByName(name: string, deep?: boolean): DisplayObject
 {
-    for (let i = 0; i < this.children.length; i++)
+    for (let i = 0, j = this.children.length; i < j; i++)
     {
         if (this.children[i].name === name)
         {
             return this.children[i];
         }
     }
 
+    if (deep)
+    {
+        for (let i = 0, j = this.children.length; i < j; i++)
+        {
+            const child = (this.children[i] as Container);
+
+            if (!child.getChildByName)
+            {
+                continue;
+            }
+
+            const target = (this.children[i] as Container).getChildByName(name, true);
+
+            if (target)
+            {
+                return target;
+            }
+        }
+    }
+
```

Comment:
What about something like this to simplify the code a little bit more?

```suggestion
    for (let i = 0, j = this.children.length; i < j; i++)
    {
        if (this.children[i].name === name)
        {
            return this.children[i];
        }

        const child = (this.children[i] as Container);

        if (!deep || !child.getChildByName)
        {
            continue;
        }

        const target = child.getChildByName(name, deep);

        if (target)
        {
            return target;
        }
    }
```